### ISSUE TITLE
LoadBalancer:get_stats fail on first job

### ISSUE THREAD

**Issue by daskinne** [jtriley/StarCluster#240]
Remove QACCT if stats (SGEStats) is on_first_job

Will allow get_stats to return the available stats including queue status.

```
    def _get_stats(self):
        master = self._cluster.master_node
        now = self.get_remote_time()
        qatime = self.get_qatime(now)
        qacct_cmd = 'qacct -j -b ' + qatime
        qstat_cmd = 'qstat -u \* -xml'
        qhostxml = '\n'.join(master.ssh.execute('qhost -xml',
                                                log_output=True,
                                                source_profile=True,
                                                raise_on_failure=True))
##        qstatxml = '\n'.join(master.ssh.execute(qstat_cmd, log_output=True,
##                                                source_profile=True,
##                                                raise_on_failure=True))
        qacct = '\n'.join(master.ssh.execute(qacct_cmd, log_output=True,
                                             ignore_exit_status=True,
                                             source_profile=True))
        stats = SGEStats()
        stats.parse_qhost(qhostxml)
        stats.parse_qstat(qstatxml, queues=["all.q", ""])
        stats.parse_qacct(qacct, now)
        log.debug("sizes: qhost: %d, qstat: %d, qacct: %d" %
                  (len(qhostxml), len(qstatxml), len(qacct)))
        return stats
```

**Comment by daskinne** [comment]
Proposed fix:

```
def _get_stats(self):
        master = self._cluster.master_node
        now = self.get_remote_time()
        qatime = self.get_qatime(now)
        qacct_cmd = 'qacct -j -b ' + qatime
        qstat_cmd = 'qstat -u \* -xml'
        qhostxml = '\n'.join(master.ssh.execute('qhost -xml',
                                                log_output=True,
                                                source_profile=True,
                                                raise_on_failure=True))
       qstatxml = '\n'.join(master.ssh.execute(qstat_cmd, log_output=True,
                                                source_profile=True,
                                                raise_on_failure=True))
        qacct = '\n'.join(master.ssh.execute(qacct_cmd, log_output=True,
                                             ignore_exit_status=True,
                                             source_profile=True))
        stats = SGEStats()
        stats.parse_qhost(qhostxml)
        stats.parse_qstat(qstatxml, queues=["all.q", ""])
```

```
       if not stats.on_first_job():
                stats.parse_qacct(qacct, now)
```

```
        log.debug("sizes: qhost: %d, qstat: %d, qacct: %d" %
                  (len(qhostxml), len(qstatxml), len(qacct)))
        return stats
```

**Comment by daskinne** [comment]
@jtriley :+1:  This is fixed in FinchPowers vanilla improvements. Please arrange a pull for the change below to qacct. Adding handling for no history. Thanks, David

```
    def _get_stats(self):
        master = self._cluster.master_node
        now = self.get_remote_time()
        qatime = self.get_qatime(now)
        qacct_cmd = 'qacct -j -b ' + qatime
        qstat_cmd = 'qstat -u \* -xml'
        qhostxml = '\n'.join(master.ssh.execute('qhost -xml'))
        qstatxml = '\n'.join(master.ssh.execute(qstat_cmd))
        qacct = '\n'.join(master.ssh.execute(qacct_cmd, raise_on_failure=False,
                          silent=True))
        if qacct == "no jobs running since startup":
            qacct = ''
```

**Comment by jtriley** [comment]
Fixed by b3a418b58badff90d4d80e446a2c6bbceb49ca47
